interop-testing/src/main/java/io/grpc/testing/integration/AbstractTransportTest.java

```diff
@@ -586,6 +597,87 @@ public void sendsTimeoutHeader() {
     Assert.assertEquals(572000L, (long) requestHeadersCapture.get().get(ChannelImpl.TIMEOUT_KEY));
   }
 
+  @Test
+  public void deadlineNotExceeded() {
```

Comment:
I think ideally these would be in ChannelImplTest since this behavior doesn't really change based on underlying transport, but they are harder to write there and they are already here. So I don't think you need to move them, but just be aware for next time.